src/librustc_session/config.rs

```diff
@@ -2435,43 +2459,103 @@ fn parse_borrowck_mode(dopts: &DebuggingOptions, error_format: ErrorOutputType)
+            // --extern prelude_name=some_file.rlib
+            match entry {
+                Entry::Vacant(vacant) => {
+                    let mut files = BTreeSet::new();
+                    files.insert(path);
+                    vacant.insert(ExternEntry::new(ExternLocation::Files(files)))
+                }
+                Entry::Occupied(occupied) => {
+                    let ext_ent = occupied.into_mut();
+                    match ext_ent {
+                        ExternEntry { location: ExternLocation::Files(files), .. } => {
+                            files.insert(path);
+                        }
+                        _ => early_error(
+                            error_format,
+                            "--extern with `=path` cannot be specified with \
+                             --extern without `=path`",
+                        ),
+                    }
+                    ext_ent
+                }
+            }
+        } else {
+            // --extern prelude_name
+            match entry {
+                Entry::Vacant(vacant) => {
+                    vacant.insert(ExternEntry::new(ExternLocation::ExternPrelude))
+                }
+                Entry::Occupied(_) => {
+                    early_error(error_format, "--extern without `=path` can only be specified once")
+                }
+            }
+        };
+
+        let mut is_private_dep = false;
+        let mut add_prelude = true;
+        if let Some(opts) = options {
+            if !is_unstable_enabled {
+                early_error(
+                    error_format,
+                    "the `-Z unstable-options` flag must also be passed to \
+                     enable `--extern options",
+                );
+            }
+            for opt in opts.split(',') {
+                match opt {
+                    "priv" => is_private_dep = true,
+                    "noprelude" => {
+                        if let ExternLocation::Files(_) = &entry.location {
+                            add_prelude = false;
+                        } else {
+                            early_error(
+                                error_format,
+                                "the `noprelude` --extern option requires a file path",
+                            );
+                        }
+                    }
+                    _ => early_error(error_format, &format!("unknown --extern option `{}`", opt)),
+                }
+            }
+        }
 
-        // Crates start out being not private,
-        // and go to being private if we see an '--extern-private'
-        // flag
-        entry.is_private_dep |= private;
+        // Crates start out being not private, and go to being private `priv`
+        // is specified.
+        entry.is_private_dep |= is_private_dep;
+        // Crates start out adding to the extern prelude, but are removed if
+        // `noprelude` is specified.
+        entry.add_prelude &= add_prelude;
```

Comment:
I went ahead and switched it.  It does feel a little awkward being different from `priv`, but probably makes more sense.  In practice I'm not sure it will matter either way.